src/middleware/rewriteUrls.coffee

```diff
@@ -0,0 +1,104 @@
+url = require 'url'
+winston = require 'winston'
+Q = require 'q'
+
+Channel = (require '../../lib/model/channels').Channel
+config = require '../config/config'
+routerConf = config.get 'router'
+utils = require '../../lib/utils'
+router = require '../../lib/middleware/router'
+
+invertPathTransform = (pathTransform) ->
+  return pathTransform.replace /s\/(.*?)\/(.*?)($|\/(.*)$)/, 's/$2/$1/$4'
```

Comment:
To be honest inverting the transform confused me a bit. Esp. when reading the tests. The transform is definitely useful but maybe it would be clearer "un-inverted"? (verted? :) )

I think I understand the reasoning - since it's `s/from/to/` it helps to have `to` last since it's the `to` server - but maybe we should think about the transform differently? Perhaps we should rather think of it as `s/regex/replacement/`?